packages/ptokens-assets-evm/src/ptokens-evm-provider.ts

```diff
@@ -81,14 +62,24 @@ export class pTokensEvmProvider implements pTokensAssetProvider {
    * @param _gasPrice - The desired gas price to be used when sending transactions.
    * @returns The same provider. This allows methods chaining.
    */
-  setGasPrice(_gasPrice: number) {
+  setGasPrice(_gasPrice: bigint) {
```

Comment:
Why not accepting `bigint | boolean | number | string`? Maybe drop boolean. These are the accepted values for `BigInt()`.
Internally, treat _gasPrice as a biginteger, thus `this._gasPrice = BigInt(_gasPrice)`